### PR TITLE
Ontology term(s) properties JSON shape (#30)

### LINKED ISSUE #30: Add ontology term(s) properties JSON shape
If an LDTab table is loaded into Nanobot, we should be able to get a ontology-browser-style representation of a set of subject. First we need a JSON representation of the subject, then we'll use that to generate plain text and HTML views.

Our gadget and gizmos libraries did a pretty good job of the final HTML view that we want, but this time we're focused on using our Rust libraries, especially wiring.rs.

The representation of the subject has two parts: hierarchy and properties. This issue is about properties.

The properties view includes annotations and axioms, datatype and language tags for literals, OWL axiom annotations and RDF reifications, hyperlinks for IRIs and ontology terms, and Manchester syntax with hyperlinks.

The JSON has to carry various context with it, including all prefixes and labels. Ideally we could convert the HTML+RDFa, but that is an additional complexity that we'll worry about later.

Here's a first draft of what the results for OBI:0000070 'assay' should look like (in part), which should be close to what's already in `wiring.rs`:

```json
{
  "@prefixes": {
    "OBI": "http://purl.obolibrary.org/obo/OBI_"
  },
  "@labels": {
    "OBI:0000011": "planned process"
  },
  "OBI:0000070": {
    "rdfs:label": [
      {"object": "assay", "datatype": "@en"}
    ],
    "rdfs:subClassOf": [
      {"object": "OBI:0000011", "datatype": "_IRI"}
    ]
  },
}
```

Or we could embed the labels and expand the IRIs, which would be redundant but easier to process into HTML:

```json
{
  "OBI:0000070": {
    "rdfs:label": [
      {"object": "assay", "datatype": "@en"}
    ],
    "rdfs:subClassOf": [
      {"object": "OBI:0000011", "datatype": "_IRI", "label": "planned process", "iri": "http://purl.obolibrary.org/obo/OBI_0000011"}
    ]
  },
}
```

Or we could support both representations, and provide a function to transform the first into the second.

### LINKED ISSUE #35: Allow custom ordering of predicates
Building on #30 and the design in <https://github.com/ontodev/gizmos#predicates> we need to allow for custom ordering of predicates in the display. For example, we usually want synonyms first, then the definition and definition source, then whatever else, then some specific annotations at the end of the list. OLS puts logical predicates at the end, and we might want to do the same sometimes.

### PR DESCRIPTION
**Pull request by ckindermann**
Closes #30 #35

### CODE REVIEW

**Comment by ckindermann** [comment]
Outstanding TODOs:
- Doc string code examples (most of the functions take an LDTab database as input -- maybe we should create a small example database and put it in `src/resource`)
- Error handling would need to be revised at some point. An Error is returned for one of two reasons: either, a query to a database returns an `sqlx::Error` which we can just propagate, or an `unwrap` to a `serde::Value` doesn't work because an LDTab data entry is incorrect or not handled correctly. I think it would be best to introduce Rust types for the LDTab data format (doing this isn't easy though -- and this might not be a good idea if the LDTab data format is going to change).

**Comment by jamesaoverton** [comment]
Ok. Please add an example database and some tests to demonstrate the functionality, then let me know.

File: tests/term_property_json_shape.rs [hunk 1]
```diff
@@ -0,0 +1,781 @@
...
+
+    let property_map = get_property_map(&subject, &table, &pool).await.unwrap();
+    let expected_property_map = json!(
+    {
+      "obo:IAO_0000115": [
+        {
+          "object": "Compound organ that consists of gill filaments, gill lamellae, gill rakers and pharyngeal arches 3-7. The gills are responsible for primary gas exchange between the blood and the surrounding water.",
+          "datatype": "xsd:string",
+          "annotation": {
+            "<http://www.geneontology.org/formats/oboInOwl#hasDbXref>": [
```
Comment (jamesaoverton) [inline tests/term_property_json_shape.rs]: Why is this an IRI and not a CURIE? I see the `oboInOwl` prefix in the `prefix` table of the SQLite database.

File: tests/term_property_json_shape.rs [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+              {
+                "datatype": "xsd:string",
+                "meta": "owl:Axiom",
+                "object": "http:http://www.briancoad.com/Dictionary/DicPics/gill.htm"
```
Comment (jamesaoverton) [inline tests/term_property_json_shape.rs:72]: Is this `http:http://...` in the source ontology? I assume so, but I can't see it in OLS or Ontobee.

Comment (jamesaoverton) [reply tests/term_property_json_shape.rs:72]: Now I see it, nevermind.

File: tests/term_property_json_shape.rs (hunk 1 above)
Comment (jamesaoverton) [reply tests/term_property_json_shape.rs]: I expect that the `oboInOwl` prefix wasn't actually used by LDTab when the SQLite file was created. I pushed a commit that fixes this.

Comment (ckindermann) [reply tests/term_property_json_shape.rs]: That's right. The `oboInOwl` prefix wasn't configured when the SQLite file was created. We probably should have dedicated tests for both IRIs and CURIEs. Even though they are both simply treated as strings in the code and should give rise to the same behaviour, this is not always the case. The function `get_prefixes` [here](https://github.com/ontodev/nanobot.rs/blob/term-properties-json/src/ldtab.rs#L53) is an example. So far, I didn't encounter any issues with this. But the angle brackets used in IRIs might cause issues in the HMTL encoding.

**Pull request merged**